spec/actors/hyku_addons/actors/creator_display_profile_visibility_actor_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe HykuAddons::Actors::CreatorProfileVisibilityActor do
+  let(:creator) do
+    [
+      {
+        creator_name_type: "Personal",
+        creator_family_name: "Hawking",
+        creator_given_name: "Stephen",
+        creator_institutional_email: user.email,
+        creator_profile_visibility: User::PROFILE_VISIBILITY[:closed]
+      }
+    ].to_json
+  end
+  let(:work) { create(:work, creator: [creator]) }
+  let(:account) { create(:account) }
+  let(:site) { Site.new(account: account) }
+
+  # Middleware Setup
+  let(:attributes) { work.attributes }
+  let(:user) { create(:user) }
+  let(:ability) { ::Ability.new(user) }
+  let(:env_class) { Hyrax::Actors::Environment }
+  let(:env) { env_class.new(work, ability, attributes) }
+  let(:terminator) { Hyrax::Actors::Terminator.new }
+  let(:middleware) do
+    stack = ActionDispatch::MiddlewareStack.new.tap do |middleware|
+      middleware.use described_class
+    end
+
+    stack.build(terminator)
+  end
+  let(:key) { "creator_profile_visibility" }
+
+  before do
+    allow(Site).to receive(:instance).and_return(site)
+  end
+
+  describe "#create" do
```

Comment:
There are no tests here where the email fails validation.  We need several scenarios where the email doesn't match the required format, is nil, blank or contains characters which are invalid.

Reply:
Sorry ignore this comment, a test for that should be in the user behavior spec